File: Initial Implementation/CaesarCipher.java
public class CaesarCipher {

    public static void main(String[] args) {
        final String alphabet = args[0];
        final String alphabetLowercase = alphabet.toLowerCase();
        final String text = args[1];

        System.out.println("\nАлфавіт:\n" + alphabet + "\n" + alphabetLowercase + "\n");
        System.out.println("Оригінал:\t" + text);

        for (int shift = 1; shift < alphabet.length(); shift++) {
            StringBuilder result = new StringBuilder();
            for (char textChar : text.toCharArray()) {
                String textCharStr = String.valueOf(textChar);
                if (alphabet.contains(textCharStr)) {
                    result.append(shiftChar(textChar, shift, alphabet));
                } else if (alphabetLowercase.contains(textCharStr)) {
                    result.append(shiftChar(textChar, shift, alphabetLowercase));
                } else {
                    result.append(textChar);
                }
            }
            System.out.println("Зсув = " + shift + ":\t" + result);
        }
        System.out.println();
    }

    private static char shiftChar(char c, int shift, String alphabet) {
        return alphabet.charAt(
                (alphabet.indexOf(c) + shift) % alphabet.length()
        );
    }
}

File: src/main/kotlin/CaesarCipher.kt
class CaesarCipher {
    companion object {
        @JvmStatic
        fun decode(alphabet: String, encodedText: String): List<String> {
            val alphabetLowercase = alphabet.lowercase()

            return (1..alphabet.length).map { shift ->
                encodedText.asSequence()
                    .map {
                        if (alphabet.contains(it)) {
                            shiftChar(it, shift, alphabet)
                        } else if (alphabetLowercase.contains(it)) {
                            shiftChar(it, shift, alphabetLowercase)
                        } else {
                            it
                        }
                    }
                    .joinToString("")
            }
        }

        private fun shiftChar(c: Char, shift: Int, alphabet: String): Char {
            return alphabet[
                    (alphabet.indexOf(c) + shift) % alphabet.length
            ]
        }
    }
}